app/component/ItinerarySummaryListContainer/ItinerarySummaryListContainer.js

```diff
@@ -2,22 +2,27 @@ import PropTypes from 'prop-types';
 import React, { useState } from 'react';
```

Comment:
Good points!

In this PR, `index.js` file keeps the import statement untouched.

Splitting big component to smaller components with more strict responsibilities yields many components and I think it's natural to store these hierarchially rather than as a flat list and let the directory layout help with code-base navigation. Eg. SummaryPage, apporx. 2400 LOC probably could be splitted to dozens of child components. In hsl.fi project there's a good example of using nesting with [CityBikesRentalsBlock](https://dev.azure.com/hslfi/hslfi/_git/hslfi-frontend?path=%2FFrontend%2Fsrc%2Fblocks%2FCityBikesRentalsBlock&version=GBdevelop&_a=contents) (they seem to use one index.ts file per directory). This nested hierarchy definitely needs some rules simplify code-base navigation rather than making it more complex. Natural basic rule is that components should only import components from *subdirectory or from parent directories*, but *not from any parent's subdirectory*. This way generic components would live higher in the directory hierarchy. Also it would enable extracting a child component without need to worry about adding noise/complexity to directory listings.

I think that classification simple and pure utility components from more fat container like orchestrator components boils down to a debated question of how to differentiate [presentational vs container components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0). It seems that the current practice is not to make this separation and just treat all components equally.

Just wondering should we run a single automated batch operation to convert all current `app/components/Xxx.js` to `app/component/Xxx/{Xxx.js,index.js}` structure to make project layout homogenous?

Reply:
I'm not sure if the file `findErrorMessageIds.js` is either presentational or container component as it just contains utility functions but no react.

I think we first try use this directory structure for a bit and see if we find any issues with it, if not, then we could do the refactoring to change all the components to be in directories.

Reply:
Indeed, I think `utils` subdirectory will work fine. 👍  I read the comment too quickly and not properly checked the exact files in question, excuse my misunderstanding.